gammapy/scripts/analysis.py

```diff
@@ -242,43 +241,56 @@ def _create_geometry(params):
+        stack_datasets = self.settings["reduction"]["stack-datasets"]
 
-        if "offset-max" in self.settings["reduction"]:
-            maker_params["offset_max"] = Angle(self.settings["reduction"]["offset-max"])
+        if stack_datasets:
+            stacked = MapDataset.create(geom=geom, geom_irf=geom_irf, name="stacked")
 
-        geom = self._create_geometry(self.settings["reduction"]["geom"])
+            for obs in self.observations:
+                dataset = self._get_dataset(obs, geom, geom_irf, offset_max)
+                stacked.stack(dataset)
 
-        pars = self.settings["reduction"].get("geom-irf")
-        if pars is not None:
-            geom_true = self._create_geometry(pars)
+            self._extract_irf_kernels(stacked)
+            datasets = [stacked]
         else:
-            geom_true = geom
+            datasets = []
+            for obs in self.observations:
+                dataset = self._get_dataset(obs, geom, geom_irf, offset_max)
+                self._extract_irf_kernels(dataset)
+                datasets.append(dataset)
 
-        maker = MapMaker(geom=geom, geom_true=geom_true, **maker_params)
-        maps = maker.run(self.observations)
-        table_psf = make_mean_psf(self.observations, geom.center_skydir)
-        psf_kernel = PSFKernel.from_table_psf(table_psf, geom_true, max_radius=Angle("0.3 deg"))
+        self.datasets = Datasets(datasets)
 
-        e_reco = geom.get_axis_by_name("energy").edges
-        e_true = geom_true.get_axis_by_name("energy").edges
-        edisp = make_mean_edisp(
-            self.observations, position=geom.center_skydir, e_true=e_true, e_reco=e_reco
+    @staticmethod
+    def _get_dataset(obs, geom, geom_irf, offset_max):
+        position, width = obs.pointing_radec, 2 * offset_max
+        geom_cutout = geom.cutout(position=position, width=width)
+        geom_irf_cutout = geom_irf.cutout(position=position, width=width)
+
+        maker = MapMakerObs(
+            observation=obs,
+            geom=geom_cutout,
+            geom_true=geom_irf_cutout,
+            offset_max=offset_max
         )
 
-        background_model = BackgroundModel(maps["background"], norm=1.0)
-        background_model.parameters["tilt"].frozen = False
-
-        self.datasets = Datasets(
-            MapDataset(
-                counts=maps["counts"],
-                exposure=maps["exposure"],
-                background_model=background_model,
-                psf=psf_kernel,
-                edisp=edisp,
-                name="stacked",
-            )
-        )
+        return maker.run()
+
+    @staticmethod
+    def _extract_irf_kernels(dataset):
+        #TODO: handle IRF maps in fit
+        geom = dataset.counts.geom
+        geom_irf = dataset.exposure.geom
+
+        position = geom.center_skydir
+        geom_psf = geom.to_image().to_cube(geom_irf.axes)
+        dataset.psf = dataset.psf.get_psf_kernel(position=position, geom=geom_psf, max_radius="0.5 deg")
```

Comment:
shouldn't the `max_radius` value come from the settings?

Reply:
Indeed. It seems I accidentally deleted it (or rather forgot to re-introduce...) in #2397, while I meant to move it from `geometry` to `reduction`.